pkg/controller/daemon/daemoncontroller.go

```diff
@@ -994,32 +990,37 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 
 	if ds.DeletionTimestamp != nil || !dsc.expectations.SatisfiedExpectations(dsKey) {
 		// Only update status.
-		return dsc.updateDaemonSetStatus(ds, hash)
+		return dsc.updateDaemonSetStatus(ds, hash, nodeList)
 	}
 
-	err = dsc.manage(ds, hash)
+	err = dsc.manage(ds, hash, nodeList)
 	if err != nil {
 		return err
+	} else {
```

Comment:
also @kargakis any insight on this one ? so we can move forward with this one

ping @janetkuo @enisoc